### PR TITLE
[VEN-1217]: VIP simulation Liquidator

### PR DESCRIPTION
**Pull request by GitGuru7**
Resolves VEN-1217

### CODE REVIEW

File: vips/vip-liquidator.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [LIQUIDATOR, "resumeForceVAILiquidate()", CRITICAL_TIMELOCK],
+      },
+
+      {
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [LIQUIDATOR, "setPendingRedeemChunkLength(uint256)", NORMAL_TIMELOCK],
+      },
```
Comment (chechu) [inline vips/vip-liquidator.ts:124]: Just a reminder to add the command to upgrade the implementation of the Liquidator with the new contract. It should be something like this:

```
0x2b40B43AC5F7949905b0d2Ed9D6154a8ce06084a.upgrade("0x0870793286aaDA55D39CE7f82fb2766e8004cF43", "<TODO: new Liquidator implementation>")
```

where:
- 0x0870793286aaDA55D39CE7f82fb2766e8004cF43: Liquidator proxy
- 0x2b40B43AC5F7949905b0d2Ed9D6154a8ce06084a: ProxyAdmin contract, owned by 0x939bd8d64c0a9583a7dcea9933f7b21697ab6396
- 0x939bd8d64c0a9583a7dcea9933f7b21697ab6396: Timelock contract, used in the Venus Governance process. It allows votes for 24 hours, and then it has a delay of 48 hours until the proposed VIP is executed

@GitGuru7

File: vips/vip-liquidator.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+      // {
+      //   target: PROXY_ADMIN,
+      //   signature: "upgrade(address,address)",
+      //   params: [LIQUIDATOR, "NEW_IMPLEMENTATION_ADDRESS"],
+      // },
```
[the remark this answers is not in the document]
Comment (GitGuru7) [reply vips/vip-liquidator.ts:131]: Once Liquidator changes are final, we will deploy new implementation add here

**Review by chechu — COMMENTED** [review]
testnet VIP lgtm

File: vips/vip-155/vip-155.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        target: LIQUIDATOR,
+        signature: "setPendingRedeemChunkLength(uint256)",
+        params: [10],
+      },
+
+      {
+        target: LIQUIDATOR,
+        signature: "resumeForceVAILiquidate()",
+        params: [],
+      },
```
Comment (chechu) [inline vips/vip-155/vip-155.ts:145]: I think we could postpone the activation of this feature, until every Liquidator is aware of it. So, I would remove this command from the VIP

File: simulations/vip-155/vip-155/simulations.ts [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+      comptroller = new ethers.Contract(UNITROLLER, COMPTROLLER_ABI, provider);
+      tusd = new ethers.Contract(TUSD, IERC20_ABI, provider);
+      usdc = new ethers.Contract(USDC, IERC20_ABI, provider);
+      vaiController = new ethers.Contract(VAI_CONTROLLER, VAI_CONTROLLER_ABI, provider);
+      vai = new ethers.Contract(VAI, IERC20_ABI, provider);
+      await setMaxStaleCoreAssets(CHAINLINK, NORMAL_TIMELOCK);
+    });
+
+    it("Tusd Liquidation and reduce liquidation reserves", async () => {
+      // Reserves reduced to treasury as redeem action is ative
```
Comment (chechu) [inline simulations/vip-155/vip-155/simulations.ts:221]:
```suggestion
      // Reserves reduced to treasury as redeem action is active
```

File: simulations/vip-155/vip-155/simulations.ts [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+      const protocolBalBefore = await usdc.balanceOf(TREASURY);
+      await oracle.connect(impersonatedTimelock).setDirectPrice(BTCB, parseUnits("1", 5));
+      await tusd.connect(liquidatorSigner).approve(liquidator.address, "6156967120");
+      await liquidator.connect(liquidatorSigner).liquidateBorrow(VTUSD, USER, "6156967120", VUSDC);
+      const protocolBalAfter = await usdc.balanceOf(TREASURY);
+      expect(protocolBalAfter).greaterThan(protocolBalBefore);
+    });
+
+    it("Tusd Liquidation and reduce reserves fails; action paused", async () => {
+      // Reserves will not reduce to treasury as redeem action is active
```
Comment (chechu) [inline simulations/vip-155/vip-155/simulations.ts:237]:
```suggestion
      // Reserves will not reduce to treasury as redeem action is paused
```

Comment (GitGuru7) [reply simulations/vip-155/vip-155/simulations.ts]: [6224bff](https://github.com/VenusProtocol/vips/pull/31/commits/6224bff4eb3f3d178403ada688f3ecb9906a6ad0)

File: simulations/vip-155/vip-155-testnet/simulations.ts [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+      ).equals(true);
+    });
+
+    it("Permissions addToAllowlist", async () => {
+      expect(
+        await accessControlManager
+          .connect(impersonatedLiquidator)
+          .isAllowedToCall(NORMAL_TIMELOCK, "addToAllowlist(address,address)"),
+      ).equals(true);
+    });
```
Comment (chechu) [inline simulations/vip-155/vip-155-testnet/simulations.ts:104]: This check seems to be duplicated

File: vips/vip-155/vip-155.ts [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+      {
+        target: LIQUIDATOR,
+        signature: "setPendingRedeemChunkLength(uint256)",
+        params: [10],
+      },
+
+      {
+        target: LIQUIDATOR,
+        signature: "setMinLiquidatableVAI(uint256)",
+        params: [parseUnits("100", 18)],
```
Comment (chechu) [inline vips/vip-155/vip-155.ts:144]:
```suggestion
        params: [parseUnits("1000", 18)],
```

**Review by chechu — APPROVED** [review]

**Review by Debugger022 — APPROVED** [review]

**Review by chechu — APPROVED** [review]

**Comment by chechu** [comment]
https://app.venus.io/#/governance/proposal/258?chainId=56

**Pull request merged**